app/angular/src/client/preview/types.ts

```diff
@@ -32,4 +32,6 @@ export interface StoryFnAngularReturnType {
   moduleMetadata?: NgModuleMetadata;
   template?: string;
   styles?: string[];
+  /** Uses legacy angular rendering engine that use dynamic component */
+  useLegacyRendering?: boolean;
```

Comment:
WDYT about making this a framework option. User would write the following in `.storybook/main.js`:

```js
module.exports = {
  angularOptions: {
    legacyRendering: true,
  }
}
```

Another option would be to use story parameters if we want to control this on a per story basis. To configure globally, `.storybook/preview.js`:

```js
export const parameters = {
  angularLegacyRendering: true,
}
```

Both of these would be more "storybook-like" than adding a new field to the story function return type.

Reply:
I like the idea of the framework option, because that projects rendering would be consistent, instead of potentially introducing bugs from the previous renderer not getting cleaned up.

As long as the renderer's clean themselves up when switching, then users with a large project may have a reason to migrate a few stories at a time. So, a parameter may be the better choice to avoid users not being able to start migrating, because of a bug or situation we didn't consider.